src/Tuura/Boolean.hs

```diff
@@ -0,0 +1,86 @@
+module Tuura.Boolean (
+  module Tuura.Boolean.Parser,
+  CNF, DNF, Literal (..),
+  convertToCNF, genConcepts,
+  simplifyCNF, simplifyDNF, convertCNFtoDNF) where
+
+import Tuura.Boolean.Parser
+import Data.List
+import Data.List.Extra
+import Data.Foldable
+import Data.Maybe
+import Data.Ord
+
+type CNF a = [[Literal a]]
+
+type DNF a = [[Literal a]]
+
+data Literal a = Literal { variable :: a, polarity :: Bool } deriving (Eq, Ord)
+
+convertToCNF :: Eq a => (Expr a) -> CNF a
+convertToCNF expr = cnf
+  where
+    vars = toList expr
+    values = mapM (const [False, True]) vars
+    fs = filter (\v -> not $ eval expr (\x -> getValue x vars v)) values
+    genVar val var = Literal var val
+    sim = simplifyCNF $ map (\v -> nub $ (map (\f -> (genVar (v !! f) (vars !! f))) [0..(length vars - 1)])) fs
```

Comment:
I think you shouldn't do the simplification as part of conversion to CNF. Let's separate these two concerns.